src/dwtools/js/code.js

```diff
@@ -57,38 +56,30 @@
 
         //Adding buttons
 
-        var subject = jQuery("#subject");
-        if (subject.length == 0) {
-            isPreviewPage = true;
-            $textBox = jQuery("textarea.textbox");
-            $textBox.focusout(saveSelection);
-            $textBox.bind("beforedeactivate", function () {
-                saveSelection();
-                $textBox.unbind("focusout");
-            });
-        }
-        else {
-            $textBox = jQuery("#body, #commenttext");
-            $textBox.focusout(saveSelection);
-            $(document).on("beforedeactivate", "#body", function () {
-                saveSelection();
-                $textBox.unbind("focusout");
-            });
-        }
-
-
-        for (var x = 0; x < DT['CUSTOMBUTTONS'].length; x++) {
-
-            if (isPreviewPage) {
-
-                jQuery('input[name=subject]').after(`<input type="button" data-id="${x}" class="custom-button" value="${DT['CUSTOMBUTTONS'][x].label}">`);
-            }
-            else {
-                subject.after(`<input type="button" class="custom-button" data-id="${x}" value="${DT['CUSTOMBUTTONS'][x].label}">`);
-            }
-
+        // Respectively: quick-reply, old talkform, old preview form.
+        $textBox = jQuery("#body, #commenttext, #previewform textarea.textbox");
+        $textBox.focusout(saveSelection);
+
+        var makeButton = (button, i) => `<input type="button" data-id="${i}" class="custom-button" value="${button.label}" />`;
+        var buttons =
+            '<div class="dwtools-custom-buttons">' +
+            DT['CUSTOMBUTTONS'].map( makeButton ).join(' ') +
+            '</div>';
+
+        // Try to insert between the subject and the body.
+        if ( jQuery('.qr-body').length > 0 ) {
+            // quick-reply or post-#2480 talkform. Insert above body's parent
+            // div. In mid-2019 quick-reply, this ends up above the subject, to
+            // avoid a stretchy flexbox situation. After #2522, it goes between.
```

Comment:
```suggestion
            // avoid a stretchy width: 100% situation. After #2522, it goes between.
```

whoops, there I go telling lies in the comments again